static/app/bootstrap/processInitQueue.tsx

```diff
@@ -16,6 +16,30 @@ const COMPONENT_MAP = {
 };
 
 async function processItem(initConfig: OnSentryInitConfiguration) {
+  /**
+   * Allows auth pages to dymanically load and execute Google reCaptcha V3 JavaScript API
+   */
+  if (initConfig.name === 'recaptcha') {
+    const captchaElement = document.getElementById(
+      initConfig.element_id
+    ) as HTMLInputElement;
+    const shouldAddCaptcha = captchaElement?.getAttribute(
+      initConfig.should_add_recaptcha_attr_name
+    );
```

Comment:
You should be able to use `captchaElement.dataset.shouldAddRecaptcha` as [data attributes are exposed](https://developer.mozilla.org/en-US/docs/Learn/HTML/Howto/Use_data_attributes) by the browser.